src/text-input/src/TextInput.js

```diff
@@ -12,33 +11,32 @@ const TextInput = memo(
       required,
       placeholder,
       width = 280,
-      height = 32,
       disabled = false,
       isInvalid = false,
       spellCheck = true,
+      appearance = 'default',
+      size = 'medium',
       ...restProps
     } = props
 
-    const themedClassName = useInputAppearance()
+    const { className: themedClassName, boxProps } = useInputAppearance({ appearance, size })
+    const height = restProps.height || boxProps.height
 
     return (
-      <Text
+      <Box
         is="input"
         className={cx(themedClassName, className)}
         type="text"
-        size={300}
         width={width}
         height={height}
         required={required}
         disabled={disabled}
         placeholder={placeholder}
-        paddingLeft={Math.round(height / 2.6)}
-        paddingRight={Math.round(height / 2.6)}
-        borderRadius={minorScale(1)}
         spellCheck={spellCheck}
         aria-invalid={isInvalid}
         {...(disabled ? { color: 'muted' } : {})}
```

Comment:
Oh shoot – this will fail if you aren't using `Text`

Reply:
oh good catch

Reply:
Should probably move this into the style config